### ISSUE TITLE
bug: ion-option-button should freeze list scrolling when 'swiping'

### ISSUE THREAD

**Issue by mikeacook** [driftyco/ionic#2950]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>mobile</span>  <span ionic-webview>webview</span>

<span ionic-description>I was looking for a solution for this today, and also happened upon another person with the same problem in the forums:
http://forum.ionicframework.com/t/prevent-scrolling-while-swiping-on-ion-item/14708</span>

<span is-issue-template></span>

**Comment by Ionitron** [comment]
Greetings, @mikeacook!

We are now requiring all issues to be submitted through our custom issue form so that we have all of the information necessary to fix them as quickly as possible.

Your progress on this issue has been saved at the following url:

[link to Ionic issue form](http://ionicframework.com/submit-issue/#/?iid=2950)

This issue will be closed if not resubmitted through the form. Please fill out the rest of the form so that we may assist you.

Thank you for allowing me to assist you.

**Comment by huttj** [comment]
Having this problem too.

I managed to work around it by using `overflow-scroll="true"` on the `content` tag, as well as adding in a second condition in the `_startDrag` method:

    // Or check if this is a swipe to the side drag
    else if ((!this._didDragUpOrDown && (e.gesture.direction == 'left' || e.gesture.direction == 'right') && Math.abs(e.gesture.deltaX) > 1) || (this._didDragUpOrDown && Math.abs(e.gesture.deltaY) < 5 && Math.abs(e.gesture.deltaX) > 2)) {

Before, the items would not respond, and it was hard to figure out why. Turns out it was `_didDragUpOrDown` being true, although the intended action was not to scroll, but to reveal the option buttons. Adding a second check with a cutoff for the vertical displacement allows just enough leeway for vertical thumb wobbling, without compromising the scrolling behavior.

In fact, it might be possible to simplify the check to include only the horizontal and vertical displacement, like so:

    else if (Math.abs(e.gesture.deltaY) < 5 && Math.abs(e.gesture.deltaX) > 1) {

Seems to work without any adverse effects.
